### PR TITLE
Support optional talk priority

### PR DESCRIPTION
**Pull request by ihorserba**
Needed for https://github.com/zelloptt/zellowork.com/issues/2480

### CODE REVIEW

**Comment by ihorserba** [comment]
Once approved - I'm going to add the recompiled SDK to this PR

File: sdks/js/src/classes/outgoingMessage.js [hunk 1]
```diff
@@ -164,7 +164,19 @@ outgoingMessage.then(function(result) {
...
+    if (this.options.talkPriority) {
+      switch (this.options.talkPriority) {
+        case Constants.TALK_PRIORITY_NORMAL:
+          params.talk_priority = Constants.TALK_PRIORITY_VALUE_NORMAL;
+          break;
+        case Constants.TALK_PRIORITY_LOW:
+          params.talk_priority = Constants.TALK_PRIORITY_VALUE_LOW;
+          break;
+        default:
+          params.talk_priority = this.options.talkPriority;
```
Comment (vocoded) [inline sdks/js/src/classes/outgoingMessage.js:178]: I have mixed feelings about the value being either a special string or a numeric, I would recommend instead we define `TALK_PRIORITY_NORMAL` and `TALK_PRIORITY_LOW` as the constant number values that clients of the SDK can use directly, which will allow you to remove this special case logic.

Comment (ihorserba) [reply sdks/js/src/classes/outgoingMessage.js]: Changed.
Please take a look one more time.

File: sdks/js/src/classes/outgoingMessage.js [hunk 2]
```diff
@@ -164,7 +164,10 @@ outgoingMessage.then(function(result) {
       'packet_duration': this.options.encoderFrameSize
     };
     if (this.options.for) {
-      params.for = this.options.for
+      params.for = this.options.for;
+    }
+    if (this.options.talkPriority) {
```
Comment (vocoded) [inline sdks/js/src/classes/outgoingMessage.js:169]: `0` is a valid talk priority, so we should tighten this up to `this.options.talkPriority !== undefined`. if someone puts a non-numeric value there, the API will reject it, so i don't think we need to be overly defensive here

**Review by vocoded — APPROVED** [review]
one additional change request, otherwise good

File: sdks/js/src/classes/outgoingMessage.js (hunk 2 above)
Comment (ihorserba) [reply sdks/js/src/classes/outgoingMessage.js]: Fixed

**Pull request merged**